cmd/config.go

```diff
@@ -130,41 +114,40 @@ func getConfig(flags *pflag.FlagSet) (Config, error) {
 	}, nil
 }
 
-// Reads a configuration file from disk.
-func readDiskConfig(fs afero.Fs) (Config, *configdir.Config, error) {
-	if configFile != "" {
-		data, err := ioutil.ReadFile(configFile)
-		if err != nil {
-			return Config{}, nil, err
-		}
-		var conf Config
-		err = json.Unmarshal(data, &conf)
-		return conf, nil, err
+// Reads a configuration file from disk and returns it and its path.
+func readDiskConfig(fs afero.Fs) (Config, string, error) {
+	realConfigFilePath := configFilePath
+	if realConfigFilePath == "" {
+		// The user didn't specify K6_CONFIG or --config, use the default path
+		realConfigFilePath = defaultConfigFilePath
 	}
 
-	cdir := configDirs.QueryFolderContainsFile(configFilename)
-	if cdir == nil {
-		return Config{}, configDirs.QueryFolders(configdir.Global)[0], nil
+	// Try to see if the file exists in the supplied filesystem
+	if _, err := fs.Stat(realConfigFilePath); err != nil {
+		if os.IsNotExist(err) && configFilePath == "" {
+			// If the file doesn't exist, but it was the default config file (i.e. the user
+			// didn't specify anything), silence the error
+			err = nil
+		}
+		return Config{}, realConfigFilePath, err
 	}
-	data, err := cdir.ReadFile(configFilename)
+
+	data, err := afero.ReadFile(fs, realConfigFilePath)
 	if err != nil {
-		return Config{}, cdir, err
+		return Config{}, realConfigFilePath, err
 	}
 	var conf Config
 	err = json.Unmarshal(data, &conf)
-	return conf, cdir, err
+	return conf, realConfigFilePath, err
 }
 
 // Writes configuration back to disk.
```

Comment:
Same as above - now it writes to the provided `afero.Fs`

Reply:
done